services/src/main/java/org/apache/unomi/services/impl/events/EventServiceImpl.java

```diff
@@ -250,6 +250,11 @@ private void getEventProperties(Map<String, Map<String, Object>> mappings, List<
         }
     }
 
+    public boolean hasEventAlreadyBeenRaised(Event event) {
+        Event pastEvent = this.persistenceService.load(event.getItemId(), Event.class);
```

Comment:
Actually the hasEvent* methods are a known performance issue because they perform searches over events which can be problematic for large event sets. However, this implementation seems better since we are only searching for a single event id.